src/api/routes/roles.test.ts

```diff
@@ -113,6 +116,109 @@ describe('roles endpoints', () => {
         });
     });
   });
+
+  describe('GET /roles', () => {
+    it('returns 401 if the caller is unauthenticated', async () => {
+      await request(app)
+        .get(`/api/v1/roles`)
+        .expect(401)
+        .expect((res) => {
+          expect(res.body.code).toEqual('access.unauthorized');
+        });
+    });
+
+    it('returns 403 if authenticated user does not have the required permission', async () => {
+      const authenticatedUser = await persistedUserWithRoleAndPermissions(
+        USER,
+        []
+      );
+
+      await request(app)
+        .get(`/api/v1/roles`)
+        .set({
+          Authorization: `Bearer ${await getTokenForUser(authenticatedUser)}`,
+        })
+        .expect(403)
+        .expect((res) => {
+          expect(res.body.code).toEqual('access.denied');
+        });
+    });
+
+    it('gets all the existing roles', async () => {
+      const role = await roleRepository.save(new Role('SOME_ROLE'));
+      const authenticatedUser = await persistedUserWithRoleAndPermissions(
+        ADMIN,
+        [LIST_ROLES]
+      );
+
+      await request(app)
+        .get(`/api/v1/roles`)
+        .set({
+          Authorization: `Bearer ${await getTokenForUser(authenticatedUser)}`,
+        })
+        .expect(200)
+        .expect((res) => {
+          expect(Array.isArray(res.body)).toBe(true);
+          const roles = res.body as ApiRole[];
+          expect(roles.find(({ name }) => name === ADMIN)?.name).toEqual(ADMIN);
+          expect(
+            roles.find(({ name }) => name === ADMIN)?.permissions
+          ).toEqual([LIST_ROLES]);
+        });
+    });
+  });
+
+  describe('GET /permissions', () => {
```

Comment:
Should this be in another file?

Reply:
As in, if there's a /permissions route, it should probably have it's own file, and it's own test?